### ISSUE TITLE
Self-hosted city-level geolocation

### ISSUE THREAD

**Issue by varac** [plausible/analytics#1809]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

First of all, thanks for this awesome software, it's really useful!

I'm using a [Dockerfile from todays's latest master (d40faf6e)](https://0xacab.org/varac/plausible-analytics/container_registry/201) because I'd like to use city-level geolocation support, and the last release is already over 2 month old and doesn't contain this new feature (btw, please release a new version soon :wink: ).
I deployed it using [my own helm chart](https://0xacab.org/varac/plausible-analytics-helm-chart), which is working fine.

But visitors only show up country-level on the map or the `country` tab, but not in the `region` or `city` tab, see attached screenshots.

I also tried with [Maxmind's GeoLite2 City DB](https://dev.maxmind.com/geoip/geolite2-free-geolocation-data) and pointing plausible to it using the `GEOLITE2_COUNTRY_DB` env var, but without success.

Is this even possible with a current self-hosted plausible instance ?

### Expected behavior

Visitors should show up in the  `region` and `city` tab.

### Screenshots

![image](https://user-images.githubusercontent.com/488213/161263893-d0f3ebd8-9f1f-46f0-8f91-3487d73c9791.png)
![image](https://user-images.githubusercontent.com/488213/161263954-b5ea0155-5fdf-4935-a39f-3284c538eb01.png)

### Environment

```markdown
- OS: PopOS 21.10
- Browser: firefox
- Browser Version: 98.0.2+build1-0ubuntu0.21.10.1
```

**Comment by ukutaht** [comment]
Hey @varac!

I will create a release for self-hosted in April with instructions how to use city-level geolocation tracking. Currently this feature is unreleased and, therefore, unsupported.

**Comment by ukutaht** [comment]
Sorry I don't have a better answer at the moment. We take some time testing features out before releasing them.

**Comment by varac** [comment]
@ukutaht Sure, that's reasonable. Looking forward to the release and thanks for taking care!

**Comment by larrybahr** [comment]
Just found plausible and I love it! Great work!
Was wondering if you plan on releasing that update soon? I checked the docs and dockee hub, but the stable release is still from a while ago.

**Comment by metmarkosaric** [comment]
Thanks for the kind words @larrybahr! Self-hosted version is not updated as frequently as the hosted version which changes daily. Think of it as a long term release to help us and help you not need to upgrade things all the time and risk breaking stuff. We stack all the new features and fixes that have been battle tested on the hosted version and release a new self-hosted version twice a year. You can learn more here: https://plausible.io/blog/building-open-source

**Comment by larrybahr** [comment]
I totally understand. I ask because ukutaht said a release would be cut in April, but the docker image is older than that.
Is there a place I can find the release schedule so as to not bother the devs with comments?

**Comment by metmarkosaric** [comment]
you can see all the releases here: https://github.com/plausible/analytics/releases. last release was on january 26th 2022. the hope to release them every 6 months but we don't have any fixed release schedule so i cannot make any promise on when the next one will be done. we're a small team with limited development resources so things take time.

**Comment by larrybahr** [comment]
Thank you for the info and your time!

**Comment by metmarkosaric** [comment]
you're welcome!

**Comment by varac** [comment]
> I will create a release for self-hosted in April with instructions how to use city-level geolocation tracking.

The last release is from Jan this year, please release a new version including this feature :pray:

**Comment by jbraithwaite** [comment]
Any update on this? Can this issue be reopened?

**Comment by bpawel-bclub** [comment]
> - GEOLITE2_COUNTRY_DB=/geoip/GeoLite2-City.mmdb # change this line

Should I also change "GEOLITE2_COUNTRY_DB" to "GEOLITE2_CITY_DB"?

Also, I have followed this, and the official [docs](https://plausible.io/docs/self-hosting), restarted the service but I still see "IP Geolocation by DB-IP" in the dashboard. What am I missing? Thank you.

**Comment by ruslandoga** [comment]
@bpawel-bclub 👋

I think the env var is either `GEOLITE2_COUNTRY_DB` or `IP_GEOLOCATION_DB`, [more info.](https://github.com/plausible/analytics/discussions/2324#discussioncomment-3862569)

> Also, I have followed this, and the official [docs](https://plausible.io/docs/self-hosting), restarted the service but I still see "IP Geolocation by DB-IP" in the dashboard. What am I missing? Thank you.

You might be running an older version, like `v1.4`.

**Comment by bpawel-bclub** [comment]
Thank you @ruslandoga ! Yes, I have just updated to rc1 and it works 🙏🏼

**Comment by woss** [comment]
For me, this doesn't work when I would like to have both county and city data displayed.

Here is the .env part
```ini
GEOLITE2_COUNTRY_DB=/geoip/GeoLite2-Country.mmdb
IP_GEOLOCATION_DB=/geoip/GeoLite2-City.mmdb
```
And this part of the docker-compose.yml

```yaml
# ...before

  plausible:
    image: plausible/analytics:v1.5.1
    restart: unless-stopped
    networks:
      - plausible
      - caddy
    labels:
      caddy: 'analytics.kelp.digital'
      caddy.reverse_proxy: '{{upstreams 8000}}'
    command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"
    volumes:
      - ./data/geoip:/geoip:ro
    depends_on:
      - plausible_db
      - plausible_events_db
      - geoip
    env_file:
      - plausible-conf.env
  geoip:
    image: maxmindinc/geoipupdate
    restart: unless-stopped
    networks:
      - plausible
      - caddy
    env_file:
      - plausible-conf.env
    volumes:
      - ./data/geoip:/usr/share/GeoIP

# ...after
```

Files are there:
![image](https://user-images.githubusercontent.com/80440/230118334-286dd216-b550-4883-90a3-ed4853ad1122.png)


And the proof that nothing is showing
![image](https://user-images.githubusercontent.com/80440/230118214-fc491cc3-3cc9-4ff7-81fa-9c6d3e8bfb76.png)

**Comment by woss** [comment]
Hi @ruslandoga I'm afraid this is all i can see 
![image](https://user-images.githubusercontent.com/80440/230142255-72929ab3-14a0-46f8-9312-7e52e7cbda2b.png)

**Comment by ruslandoga** [comment]
You'd need to remove `iex>` from the commands :) I used it to indicate that these should be executed in an iex session (which you are already in).

**Comment by woss** [comment]
HEHE i'm don't speak Elixir :) 

So good news and not-so-good news. The commands you gave still throw error 
![image](https://user-images.githubusercontent.com/80440/230158906-b73ff113-faa5-40ff-8e9b-d750f26cba4c.png)

BUT I can see the cities, although not a full list
![image](https://user-images.githubusercontent.com/80440/230159025-491b272e-e007-4163-884b-47ef794ba6dc.png)

**Comment by ruslandoga** [comment]
Oh, sorry. I forgot that `v1.5.1` is using a different library for geolocation than `master`. I think the equivalent commands would be

```elixir
Geolix.metadata()
```

and

```elixir
Geolix.lookup("8.8.8.8")
```

**Comment by woss** [comment]
![image](https://user-images.githubusercontent.com/80440/230184310-9ba796d2-abf1-4d2c-94e2-c9559151fc05.png)
![image](https://user-images.githubusercontent.com/80440/230184383-f46d546c-c15b-41b5-a67a-dbcb29bc764f.png)

**Comment by woss** [comment]
this is all the output

**Comment by ruslandoga** [comment]
It seems like the database is setup correctly.

**Comment by woss** [comment]
it seems so, so why then i see only parts of the cities? 
Here are the countries and cities screenshot

![image](https://user-images.githubusercontent.com/80440/230316069-05265d65-2a2b-4a3e-9cfd-7e91238dd524.png)
![image](https://user-images.githubusercontent.com/80440/230316104-58ba8b96-6103-46bc-a15a-a2593f4f3518.png)

**Comment by woss** [comment]
Thanks, @ruslandoga for the explanation, i've downloaded the csv file and mounted it, and created the correct entries.

I also recreated all the containers and still got the same list of cities.  

I tested the loading of the geolocation db by providing the wrong name for the csv file and as expected it failed. The correct name doesn't throw an error though.  Could it be that the data isn't saved and not referenced? for me this is not that important but for some it could be. 

![image](https://user-images.githubusercontent.com/80440/230404083-0470025f-b27d-488b-9048-a49007635875.png)
![image](https://user-images.githubusercontent.com/80440/230404199-b416fa10-e6a9-48a9-b99e-39b54ff4940e.png)
